### DISCUSSION TITLE
Auto starting at bootup

### DISCUSSION THREAD

**Discussion by Daviboi** [flyte/mqtt-io#248] [Q&A]
Hey guys, just started using this but I was wondering if there was an easy way to start this when raspbian boots up? I have tried doing an RC.local entry but I am having issues with it being a module vs a script. Any help would be amazing! Thanks, DaviBoi

**Comment by BenjiU** [comment]
Hi Daviboi,
 
in the "old" versions, there was a hint for using it with supervisor: https://github.com/flyte/mqtt-io/tree/0.5.6#serving-suggestion
This should still be valid, but you maybe have to adapt some of the commands.

Greetings!

**Comment by OzGav** [comment]
Just spent hours trying to get this to work and finally cracked it. I had it running fine from the command line but as soon as I tried all manner of ways to get it to autostart I would get `/usr/bin/python3: No module named mqtt_io`

In the end I edited /etc/rc.local and got it working with this (redirecting the output to a txt file is not strictly necessary you could use /dev/null)
`PYTHONPATH=/home/pi/.local/lib/python3.7/site-packages/ /usr/bin/python3 -m mqtt_io /home/pi/config.yml >> /tmp/mqtt-io-error.txt 2>&1`

**Reply by BenjiU** [reply]
Hmm, that sounds like a user/root-Problem: you may have installed mqtt_io in your user account, but when you try to run it via "autostart" (as root user) it can't find mqtt_io. Maybe you can try to install it with 'sudo'.
That is a hint for me: you have to set the python path to the "pi"-user location:
`PYTHONPATH=/home/pi/.local/lib/python3.7/site-packages`

**Reply by OzGav** [reply]
Quite possibly. I just followed the installation instructions here https://mqtt-io.app/2.2.6/#/?id=installation

I am actually just going to leave it as it is working fine for me but future readers may find this discussion helpful.
